### DISCUSSION TITLE
Chi square test with effect size measure in multiple imputed datasets

### DISCUSSION THREAD

**Discussion by anamgreco** [amices/mice#355] [Impute, analyse and pool]
Hi and thank you once more!

I need to perform a chi square test to compare frequencies across three groups, with a corresponding effect size measure in the multiple imputed datasets. All I found about how to pool the results in multiple imputed datasets for chi square test is here (https://bookdown.org/mwheymans/bookmi/data-analysis-after-multiple-imputation.html) 

It looks like I need to use miceadds::micombine.chisquare and manually insert the values with c(). But I am confused about the order in which I should include the values and how to let R know that I am comparing three groups across 10 imputed datasets. I am also lost in terms of how can I include a Cramer’s V estimation.

THANKS AGAIN

**Comment by stefvanbuuren** [comment]
Your question is rather general, so my answer is also generic.

If you have a parameter of scientific interest (e.g. an effect estimate), pool that by pool() or pool.scaler(). 

If you don't have a parameter, but only a statistic (chi-square, p-value) use `D2()`, knowing that method is [statistically inferior](https://stefvanbuuren.name/fimd/sec-multiparameter.html#sec:chi).

**Comment by anamgreco** [comment]
Thank you again.

I am sorry for re-opening this but I have been trying to pool this out all this time with no success. Sorry again.

I am trying to use pool() function to get the Cramer's V effect size measure comparing proportions across three groups. In a single imputed dataset I would use cramerV() from rcompanion package. Following ?pool, I applied this function using with() as below:

`imp<-mice(data, m = 10)
V<-with(imp, cramerV(X, Y)`

This gives me the 10 estimates of cramer's V for each imputed dataset. However, when I try to use pool() to get a single estimate, I get the following errors:

Error: $ operator is invalid for atomic vectors
In addition: Warning messages:
1: 'tidy.numeric' is deprecated.
See help("Deprecated") 

The second is repeated 10 times. 

I am not including any "$" in my function, so I am not sure why it says that. Do you know what could that be? 
Thank you very much for all your help!

**Comment by stefvanbuuren** [comment]
Apparently, `cramerV()` doesn't yield output that `broom` and hence `mice` can understand. The output of the `CramerV` function is very minimal, and doesn't even include a P-value. Without any measure of uncertainty, it's not really useful to apply Rubin's rules. I am not familiar with any theory that say how the pool CramerV. 

As a quick fix, take the average. If the individual CramerV's vary a lot, you know that the missing data mechanism may have a big impact on the measure.

**Comment by anamgreco** [comment]
Ok, thanks a lot! 
I have been looking for other packages to compute Cramer's v, but the output was similar. 

THANKS AGAIN =)

**Comment by anamgreco** [comment]
Hello, once more!

I am sorry to write again, but I am having a similar problem trying to compute D2. Since `cramerV()` does not provide a any significance test output, I computed the chi squared test across the three groups in each imputed dataset to assess whether if the different proportions across groups was significant. But when I try to compute D2 using the following code:

`chi<-with(imp, chisq.test(X, Y)`
`D2(chi)`

I get this output:

Error: Problem with `summarise()` input `qbar`.
x Column `estimate` not found in `.data`
i Input `qbar` is `mean(.data$estimate)`.
Run `rlang::last_error()` to see where the error occurred.
In addition: Warning messages:
1: Unknown or uninitialised column: `df.residual`. 
2: In pool.fitlist(getfit(object), dfcom = dfcom) : Large sample assumed.

I guess something similar is happening with the function `chisq.test`, but I wanted to ask whether if you (or anybody here, please) knew any other way of testing if the difference between the proportions of the three groups is significant. Looking at the p values of each chi squared test seems pretty evident but I am not sure if I should report something else…

Thank you again and apologize once more. Hopefully next time I use MI I would need less help ;)

**Comment by stefvanbuuren** [comment]
Same problem. The `cramerV` function produces output that `broom` (and hence `mice`) does not understand. You would need either
- to create a correct `tidy` function, or 
- to fill in the results by hand in formula's 5.6 - 5.9 in https://stefvanbuuren.name/fimd/sec-multiparameter.html#sec:chi. 

Sorry, no easy solution here.

**Comment by anamgreco** [comment]
Thanks again, anyway!
